### PR TITLE
Add support for Sacred

### PR DESCRIPTION
**Pull request by AyushExel**
### Details
[Sacred](url) is a tool to help you configure, organize, log and reproduce experiments. It is designed to do all the tedious overhead work that you need to do around your actual experiment. This PR adds support for `wandbObserver` class which enables native wandb logging in Sacred.
Here's a simple usage example:
```python
from numpy.random import permutation
from sklearn import svm, datasets
from sacred import Experiment

ex = Experiment('iris_rbf_svm')
from wandb.sacred import WandbObserver
ex.observers.append(WandbObserver(project='sacred_test',
                                        name='test1'))
@ex.config
def cfg():
    C = 1.0
    gamma = 0.7
@ex.automain
def run(C, gamma, _run):
    iris = datasets.load_iris()
    per = permutation(iris.target.size)
    iris.data = iris.data[per]
    iris.target = iris.target[per]
    clf = svm.SVC(C, 'rbf', gamma=gamma)
    clf.fit(iris.data[:90],
            iris.target[:90])
    return clf.score(iris.data[90:],
                        iris.target[90:])
````
The run generated by the above function can be viewed [here](https://app.wandb.ai/wandb/sacred_test/runs/3045euly)
A more detailed pytorch example and its dashboard can be viewed [here](https://app.wandb.ai/wandb/sacred_test/runs/16db8vyt)

### CODE REVIEW

File: wandb/sacred/__init__.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import collections
+import warnings
+
+from sacred.dependencies import get_digest
+from sacred.observers import RunObserver
+import numpy
+import wandb
+class WandbObserver(RunObserver):
+    """Logs sacred experiment data to Neptune.
```
Comment (vanpelt) [inline wandb/sacred/__init__.py:9]: Neptune => W&B

File: wandb/sacred/__init__.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            iris.data = iris.data[per]
+            iris.target = iris.target[per]
+            clf = svm.SVC(C, 'rbf', gamma=gamma)
+            clf.fit(iris.data[:90],
+                    iris.target[:90])
+            return clf.score(iris.data[90:],
+                                iris.target[90:])
+    """
+
+    def __init__(self, project=None, name=None):
```
Comment (vanpelt) [inline wandb/sacred/__init__.py:36]: We should add `reinit=None` to this as well, or maybe a **kwargs to just pass through options to init.

Comment (AyushExel) [reply wandb/sacred/__init__.py]: Added support for passing arguments to `wandb.init()` using `**kwargs`. 
``` python
  def __init__(self, **kwargs):
      self.run =  wandb.init(**kwargs)
```
Here's the complete pytorch test code on [W&B dashboard](https://app.wandb.ai/cayush/sacred_test/runs/19suhfj4/code)
Some snippets from the above code:
* Passing `wandb.init()` arguments in the `WandbObserver`
```python
ex.observers.append(WandbObserver(project='sacred_test',    
                                        name="Torch_test", 
                                        config={"Job_type":"torch_test","gpu_type":"t4"}, 
                                        reinit=False ))
```
* Alternatively, `WandbObserver` will automatically add elements to `wandb.config` that are present inside a function with @ex.config decorator. 
```python
@ex.config
def get_config():
    '''
    These variables are automatically tracked in W&B config dashboard
    '''
    input_size = 784
    hidden_size = 500
    num_classes = 10
    num_epochs = 5  
    batch_size = 100
    learning_rate = 0.001
    
    model_file = 'model.ckpt'
```
Here's the config section from [W&B dashboard](https://app.wandb.ai/cayush/sacred_test/runs/19suhfj4) of this project.
Screenshot from 2020-08-30 20-28-14

**Comment by vanpelt** [comment]
@AyushExel I'm merging now, can you also add this to https://github.com/wandb/client-ng?

**Pull request merged**